src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClient.java

```diff
@@ -70,4 +78,20 @@ private long stripExpiryFromBearerToken(String tokenWithBearer) {
         return expires.asLong();
     }
 
+    private void onCachedIdamTokenRemoval(
+        String jurisdiction,
+        CachedIdamToken cachedIdamToken,
+        RemovalCause cause
+    ) {
+        userDetailsCache.invalidate(cachedIdamToken.accessToken);
```

Comment:
It would be good to log this.

Reply:
I will add log with next pr with configurable cache maxsize